src/app/band-three-sensor.service.ts

```diff
@@ -0,0 +1,30 @@
+import { SensorData } from './sensor-data';
+import { Injectable, Input } from '@angular/core';
+import { Observable, of, timer } from 'rxjs';
+import { DatePipe } from '@angular/common';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BandThreeSensorService {
+
+  sensor: Observable<SensorData>;
+  timer: Observable<number> = timer(1000, 1000);
+
+  temp = 100.1;
+
+  constructor() {
+    this.sensor = new Observable(subscriber => {
+      var dataCapture = this.timer.subscribe((seconds) => {
+        const sensorData = new SensorData();
+        var num: number = Math.sin(seconds);
+        this.temp = this.temp + num;
```

Comment:
Your temperatures are 100º ± 1 while the problem description says "...operational temperature is around 20ºK but can vary by ± 10 degrees". This is an exercise but in real life you'd need to follow the specs carefully.